### PR TITLE
feat(interview/plan): wire in interview-prep's sourced-question research before Block 4

### LINKED ISSUE #2096: feat(interview/plan): wire in interview-prep.md's sourced-question WebSearch research before falling back to pattern inference
## Problem

`modes/interview-prep.md`'s Step 1 already does exactly the right thing: it runs a structured set of WebSearch queries against Glassdoor/Blind/LeetCode-discuss for a company's actually-reported interview questions, cites every sourced question, and only falls back to `[inferred from JD]` pattern-guessing when nothing real turns up. `modes/interview/practice.md` reuses that same research path reactively, mid-session, if the candidate gets caught flat-footed on thin company intel.

But `modes/interview/plan.md` — the mode that produces the time-blocked prep plan actually used right before a scheduled interview — has no equivalent step. Block 4 ("Behavioral stories") goes straight from the candidate's existing story bank to "map existing stories to likely question types," with no instruction to first check whether real, reported questions exist for this company/round type. In practice this means `plan.md`'s output defaults to pattern-inference even when a quick search would have surfaced actual candidate-reported questions — the same gap `interview-prep.md` already solved, just not wired into this mode.

Concretely hit twice in one evening: two prep docs generated via ad-hoc plan-style prep (not the full `interview-prep` pipeline) shipped with only inferred practice questions, and only got real sourced ones after a human explicitly asked for a web search pass afterward. Nothing in `plan.md` itself would have triggered that search on its own.

## Proposed fix

Add a research-check step to `modes/interview/plan.md`, ahead of (or as part of) Block 4, that makes the mode consistent with the other two rather than being the one gap:

1. **Check for existing sourced research first.** If `interview-prep/{company-slug}-{role-slug}.md` already exists (produced by an earlier `interview-prep` run), read its Step 1/Step 3 sourced questions and reuse them — never re-search work that's already been done and cited.
2. **If no prior research file exists, run `interview-prep.md`'s Step 1 WebSearch queries directly** (same query set, same citation requirement) scoped to the audience of the specific round being planned, before drafting Block 4's behavioral-story mapping.
3. **Same tagging discipline as the other two modes:** sourced questions cite their source; anything not found falls back to `[inferred from JD]`, exactly per `interview-prep.md`'s existing convention — don't invent a third label.
4. **If the search genuinely yields nothing** (obscure company, no public interview reports), say so explicitly in the plan output and proceed with JD/profile-pattern inference — same "partial-but-honest output, not perfect-or-nothing" principle already stated elsewhere in these modes.

## Why it fits

- Zero new machinery — this wires an existing, already-well-specified research path (`interview-prep.md` Step 1) into a second consumer, the same way `practice.md` already does reactively. `plan.md` gets the proactive version.
- Keeps the existing tag convention (`[inferred from JD]` / sourced-with-citation) rather than introducing a new one.
- Matches the "Never generate fake company intel" rule `plan.md` already states in its Rules section — this just makes checking for *real* intel the default first move instead of an opt-in the human has to ask for.

## Non-goals

- Not a new search provider or scraping mechanism — reuses the exact query set `interview-prep.md` already runs.
- Not mandating a full `interview-prep` run before every `plan.md` invocation — if a prep file already exists, `plan.md` reads it; if not, it runs the targeted subset of queries relevant to the specific round, not the full company-research pass.

### PR DESCRIPTION
**Pull request by Schlaflied**
## Problem

`modes/interview-prep.md`'s Step 1 already runs a structured set of WebSearch queries against Glassdoor/Blind/LeetCode-discuss for a company's actually-reported interview questions, cites every sourced question, and only falls back to `[inferred from JD]` when nothing real turns up. `modes/interview/practice.md` already reuses that same research path reactively, mid-session, if the candidate gets caught flat-footed on thin company intel.

`modes/interview/plan.md` — the mode that produces the time-blocked prep plan right before a scheduled interview — had no equivalent step. Block 4 ("Behavioral stories") went straight from the story bank to "map existing stories to likely question types," defaulting to pattern-inference even when a quick search would have surfaced real, candidate-reported questions.

## Fix

Adds a "Research check — before drafting Block 4" step to `modes/interview/plan.md`, right before the time-block template:

1. If `interview-prep/{company-slug}-{role-slug}.md` already exists (a prior `interview-prep` run), reuse its Step 1/Step 3 sourced questions directly — never re-search work already done and cited.
2. Otherwise, run `interview-prep.md`'s Step 1 WebSearch queries directly, scoped to the audience of the specific round being planned (recruiter/HR, hiring manager, or peer/technical panel).
3. Same tagging discipline as the other two modes: sourced questions cite their source, unfound ones fall back to `[inferred from JD]` — no new label invented.
4. If the search genuinely yields nothing, say so explicitly and proceed with JD/profile-pattern inference — the same partial-but-honest principle `interview-prep.md` already applies to sparse intel.

Block 4's story-mapping line and the Rules section (next to the existing "Never generate fake company intel" rule) were both updated to reference the new step, and the new step cross-references `interview-prep.md`'s Step 1 and `practice.md`'s "When company-intel is thin mid-session" section by name rather than duplicating their query tables or logic — keeping the diff small.

Also adds a self-test in `test-all.mjs` (#63) pinning: the new section header, the reuse-existing-file rule, the cross-reference to `interview-prep.md`'s Step 1 and its `[inferred from JD]` tag convention, the sparse-intel honesty fallback, and the `practice.md` cross-reference.

## Test plan

- [x] `node test-all.mjs` passes (2042 passed, 0 failed, 1 pre-existing unrelated warning)
- [x] `git status` in the worktree shows only `modes/interview/plan.md` and `test-all.mjs` changed
- [x] No changes to `modes/interview-prep.md` or `modes/interview/practice.md` — referenced, not duplicated

Closes #2096

## Summary by CodeRabbit

* **New Features**
  * Added a “research check” stage before Block 4 in interview planning, with clearer sourced-vs-inferred guidance and an honest no-results fallback.
  * Added meeting **Platform** detection (Zoom, Teams, Google Meet, or phone) from invite text and included it in interview prep instructions.
  * Introduced a new `scan:interamt` command to scrape Interamt.de job offers, with configurable search keywords.
* **Bug Fixes**
  * Improved follow-up lock stale recovery behavior to better handle orphaned recovery guards.
  * Adjusted lock reclaim behavior with a grace floor for ownerless locks.
* **Tests**
  * Added contract checks for research-check placement, contradiction handling formatting, Platform extraction, and follow-up lock “recover” guard edge cases.

### CODE REVIEW

File: test-all.mjs [hunk 1]
```diff
@@ -8959,6 +8959,64 @@ try {
...
+  ) {
+    pass('interview/plan reuses an existing interview-prep file instead of re-searching');
+  } else {
+    fail('interview/plan missing the reuse-existing-research-file rule');
+  }
+
+  if (
+    planFlat.includes('interview-prep.md`\'s "Step 1 — Research" WebSearch queries') &&
+    planFlat.includes('[inferred from JD]')
+  ) {
```
[the remark this answers is not in the document]
Comment (Schlaflied) [reply test-all.mjs:11096]: Fixed in 0aae8dd — corrected the missing opening backtick so the assertion exactly mirrors plan.md's markdown (`interview-prep.md`'s ...).

**Comment by santifer** [comment]
Same story as #2127, @Schlaflied: merging #2122 conflicted this one too. My doing.

**One file, and it's only the test file:**

```
Auto-merging test-all.mjs
CONFLICT (content): Merge conflict in test-all.mjs
```

Nothing else clashes: every mode file and script in your branch auto-merges. The clash is that several of your PRs append assertions to the same interview block in `test-all.mjs`, so they queue up behind each other one merge at a time.

**Resolution: keep both sides.** Independent assertions, no ordering constraint.

I'm sorry this is the second rebase request today on work that was already correct. Four of your PRs touch that same test block, so each merge re-conflicts the rest: that's a queue I created by holding them all and then releasing them at once, not something wrong with your branches.

The direction question that was blocking your product PRs is settled and this one was never part of it, so these merge on sight once rebased.

**Comment by santifer** [comment]
That closes the triangle, @Schlaflied: prep researches, practice reuses, and now plan checks before Block 4 instead of mapping stories blind. The sparse-intel honesty rule is the detail I appreciate most. Merged. 🚀
